cpp/arcticdb/async/tasks.hpp

```diff
@@ -315,6 +291,113 @@ struct CopyCompressedTask : BaseTask {
     }
 };
 
+// Used in arcticdb-enterprise, do not remove without checking whether it is still used there
+struct CopyCompressedInterStoreTask : async::BaseTask {
+
+    struct AllOk {};
```

Comment:
This isn't a pattern we use anywhere else to my knowledge. Could just use `std::optional`?

Reply:
I've tidied this up a bit to,

```
    using AllOk = std::monostate;
    using FailedTargets = std::unordered_set<std::string>;
    using ProcessingResult = std::variant<AllOk, FailedTargets>;
```

which I prefer over `std::optional` as the names describe the meaning of empty vs populated optional. We have a similar pattern with `StorageResult`.